File: LMS/LMS.Application/Filters/HR/EmployeeIdentity/PenaltyFilter.cs
using LMS.Application.Filters.Base;
namespace LMS.Application.Filters.HR.EmployeeIdentity
{
    public class PenaltyFilter : Filter
    {
        public List<Guid>? DepartmentIds { get; set; }
        public bool? ByIsApplied { get; set; }
        public bool? ByIsApproved { get; set; }

        public bool IsDesc { get; set; } = true;
        public PenaltyOrderBy OrderBy { get; set; }
    }
}

File: LMS/LMS.UI/huda-center/src/app/shared/filters/hr/penalty-filter.ts
import { Filter } from '../base/filter';


export enum PenaltyOrderBy {
    ByEmployeeName = 0,
    ByAmount = 1,
    ByReason = 2,
    ByDate = 3,
    ByIsApplied = 4,
    ByIsApproved = 5
}

export interface PenaltyFilter extends Filter {
    departmentIds?: string[];
    byIsApplied?: boolean | null;
    byIsApproved?: boolean | null;
    isDesc?: boolean;
    orderBy?: PenaltyOrderBy;
}
